### ISSUE TITLE
Multiple of provisioner scripts run multiple times, varies by number of scripts.

### ISSUE THREAD

**Issue by jimklo** [mitchellh/vagrant#7685]
### Vagrant version

Vagrant 1.8.5
### Host operating system

Mac OS 10.10.5
### Guest operating system

Ubuntu 14.04 LTS
### Vagrantfile

``` ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

# Change this to run jetty (and therefore, SF Web) on a different
# port. This port is used on both guest and host, and forwerded.
jetty_port = 9090

Vagrant.configure(2) do |config|

  # Our custom SF box. Built on jenkins.
  # It can also be built locally in ../sfbox.
  # If so, uncomment the box_url line below.
  config.vm.box = "sfbox"
  # config.vm.box_url = "file:sfbox.json"
  # config.vm.box_version = "0.1.52"

  vagrant_server_url = ENV['VAGRANT_SERVER_URL'] || "https://artifactory-test.cse.sri.com"
  config.vm.box_url = "#{vagrant_server_url}/artifactory/api/vagrant/Sunflower-Vagrant-local/sfbox"
  config.vm.box_version = "0.1.52"

  config.ssh.insert_key = false

  # Network setup. For now, just forward 8080, where SF web lives.
  # Note: If you use different host/guest ports, you may run into the CORS
  # issue.
  config.vm.network "forwarded_port", guest: jetty_port, host: jetty_port

  # Memory config. More may be better.
  config.vm.provider "virtualbox" do |vb|
    vb.memory = "4096"
  end

  # ---- Ontology Configuration ----------------------------------
  # Sync the ontology folder into the VM. This then gets copied
  # into the VM itself in the provisioning script.
  config.vm.synced_folder "../ontology/flora", "/opt/web/mybase/resources/ontology-source"
  #---------------------------------------------------------------

  # Set the timezone to the host timezone
  require 'time'
  offset = -Time.zone_offset(Time.now.zone)/60/60
  zonesuffix = (offset >= 0 ? "+" : "") + "#{offset.to_s}"
  timezone = 'Etc/GMT' + zonesuffix
  config.vm.provision "fix-tzone", type: "shell", inline: <<-SHELL
    if grep -q UTC /etc/timezone ; then
      echo \"#{timezone}\" | sudo tee /etc/timezone && dpkg-reconfigure -f noninteractive tzdata
    fi
  SHELL

  # Provision the box
  config.vm.provision "shell", path: "provision-sfweb.sh", args: "#{jetty_port}"

end
```
### Debug output

https://gist.github.com/jimklo/6fb7372511d9d85c7c2961e93df08939
### Expected behavior

Each provisioner script should run only once
### Actual behavior

Last provisioner script within the Vagrantfile  ran twice, while first provisioner script ran once.

Also note the same behavior occurs regardless of whether I use:

``` ruby
 config.vm.provision "fix-tzone", type: "shell", inline: <<-SHELL
    if grep -q UTC /etc/timezone ; then
      echo \"#{timezone}\" | sudo tee /etc/timezone && dpkg-reconfigure -f noninteractive tzdata
    fi
  SHELL
```

or 

``` ruby
 config.vm.provision "shell", inline: <<-SHELL
    if grep -q UTC /etc/timezone ; then
      echo \"#{timezone}\" | sudo tee /etc/timezone && dpkg-reconfigure -f noninteractive tzdata
    fi
  SHELL
```
### Steps to reproduce
1. VAGRANT_HOME=. vagrant destroy
2. VAGRANT_HOME=. vagrant up
### References

Are there any other GitHub issues (open or closed) that should be linked here?
For example:
#5518

**Comment by Pysis868** [comment]
I've been trying to recreate this, I think ever since your version!  I upgraded to 1.8.6 to avoid an SSH key bug where it would not reconnect after inserting the more secure key, and now I am at 1.9.1 getting this weird condition again.

While trying to create a MVCE with simply a box instruction, I was not able to re-create this same situation.  Maybe I can try destroying and re-creating/-provisioning that sample VM again, or try o get details from my complex `Vagrantfile` some other time..

**Comment by mickengland** [comment]
Don't know if this fix was ever released but I am seeing this same issue with the following versions:

**Vagrant Version**
$ vagrant --version
Vagrant 1.9.5

**Mac OS**
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.12.5
BuildVersion:	16F73

**Ubuntu Guest**
$ cat /etc/issue
Ubuntu 16.04.2 LTS \n \l

**Comment by briancain** [comment]
Hello @jimklo ! Thank you for reporting this issue. I just fixed this bug with #8680. The behavior you were seeing came from the Vagrantfile being loaded multiple times (likely due to having your Vagrant file in the same directory as `VAGRANT_HOME`) as well as a bug internally to Vagrant. _More information about Vagrantfile loading can be read here_: https://www.vagrantup.com/docs/vagrantfile/#load-order-and-merging

This behavior revealed an issue with the provisioner object itself and how Vagrant handles provisioners without a name. If you'd like to get this fixed without waiting for the next release of Vagrant, you can name your provision blocks and they should no longer run twice.

For example:

```ruby
 config.vm.provision "shell", path: "provision-sfweb.sh", args: "#{jetty_port}"
```

Now becomes:

```ruby
 config.vm.provision "provision_sfweb", type: "shell", path: "provision-sfweb.sh", args: "#{jetty_port}"
```

Adding the name for your provision block should prevent it from running twice. I hope this helps! Thanks :)

**Comment by jimklo** [comment]
@briancain Thanks!  I think that is how I had resolved it in the interim, by naming each provisioner.
